.github/workflows/build-image.yml

```diff
@@ -0,0 +1,166 @@
+      # https://github.com/docker/metadata-action
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@96383f45573cb7f253c731d3b3ab81c87ef81934 # v5.0.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      # Build and push Docker image with Buildx (don't push on PR)
+      # https://github.com/docker/build-push-action
+      - name: Build and push Docker image
+        id: buildandpush
+        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09 # v5.0.0
+        with:
+          context: .
+          push: false
+          load: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          
+      - name: Add Environment Variables
+        id: env
+        run: |
+          echo "GH_PULLREQ_NUM=${{ github.event.number }}" >> $GITHUB_ENV
+          echo "GH_EEVENT_NAME=${{ github.event_name }}" >> $GITHUB_ENV
+          echo "GH_RREPOSITORY=${{ github.repository }}" >> $GITHUB_ENV
+      - name: Create Tar Image For Upload
+        id: tar
+        run: |
+          docker images
+          tag=${{ steps.meta.outputs.tags }}
+          echo $tag
+          docker save -o image.tar $tag
+        
+      - name: Upload Artifact
+        id: upload
+        uses: actions/upload-artifact@v4
+        with:
+          name: image.tar
+          path: image.tar
+    outputs:
+      prnum: ${{ github.event.number }}
+      url: ${{ steps.upload.outputs.artifact-url }}
+      tag:  ${{ env.IMAGE_NAME }}
+  
+  run-lemlib:
+    needs: Build_Container
+    uses: ./.github/workflows/test-lemlib.yml
+    with:
+      GH_PR_NUM: ${{ needs.Build_Container.outputs.prnum }}
+      CALLER_TOKEN: ${{ github.token }}
+    
+  run-lemlib-no-artifact:
+    needs: Build_Container
+    uses: ./.github/workflows/test-lemlib-no-artifact.yml
+    with:
+      GH_PR_NUM: ${{ needs.Build_Container.outputs.prnum }}
+      CALLER_TOKEN: ${{ github.token }}
+
+  run-lemlib-no-hash:
+    needs: Build_Container
+    uses: ./.github/workflows/test-lemlib-no-hash.yml
+    with:
+      GH_PR_NUM: ${{ needs.Build_Container.outputs.prnum }}
+      CALLER_TOKEN: ${{ github.token }}
+  run-ez-template:
+    needs: Build_Container
+    uses: ./.github/workflows/test-ez-template.yml
+    with:
+      GH_PR_NUM: ${{ needs.Build_Container.outputs.prnum }}
+      CALLER_TOKEN: ${{ github.token }}
+  run-lemlib-copy:
+    needs: Build_Container
+    uses: ./.github/workflows/test-lemlib-with-copying-info-files.yml
+    with:
+      GH_PR_NUM: ${{ needs.Build_Container.outputs.prnum }}
+      CALLER_TOKEN: ${{ github.token }}
+  run-lemlib-errors:
+    needs: Build_Container
+    uses: ./.github/workflows/test-lemlib-with-errors.yml
+    with:
+      GH_PR_NUM: ${{ needs.Build_Container.outputs.prnum }}
+      CALLER_TOKEN: ${{ github.token }}
```

Comment:
is it possible to call these in a matrix?

Reply:
How would that work?